File: reservations - API/reservations/ProjectCalculator.Domain/Domain/Room.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Reservations.Core.Domain
{
    public class Room
    {
        public Room()
        {
            UserRooms = new List<RoomReservation>();
        }
        public string Name { get; set; }
        public Guid Id { get; set; }
        public Guid OfficeId { get; set; }
        public string Description { get; set; }
        public int Seats { get; set; }
        public bool HasTV { get; set; }
        public bool HasWhiteBoard { get; set; }
        public bool HasProjector { get; set; }
        public string OtherEquipment { get; set; }

        List<RoomReservation> UserRooms { get; set; }
        public Office Office { get; set; }
    }
}

File: reservations - SPA/reservations-spa/src/app/_models/Room.ts
import { Photo } from './Photo';

export class Room{
    public id:string;
    public name:string;
    public officeId:string;
    public description:string;
    public seats:number;
    public hasTV:boolean;
    public hasWhiteBoard:boolean;
    public hasProjector:boolean;
    public otherEquipment:string;
    public photos?:Photo[];
    public mainUrl?:string;


    constructor(id:string, name:string, officeId:string,description:string,seats:number, hasTV:boolean, hasWhiteBoard:boolean,hasProjector:boolean,
        otherEquipment:string, photos?:Photo[], mainUrl?:string){
        this.id=id;
        this.name=name;
        this.officeId=officeId;
        this.description=description;
        this.seats=seats;
        this.hasTV=hasTV;
        this.hasWhiteBoard=hasWhiteBoard;
        this.hasProjector=hasProjector;
        this.otherEquipment=otherEquipment;
        this.mainUrl = photos?.filter(x => x.isMain)[0].photoUrl;
        this.photos = photos
    }
} 